update README

voku opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/rectorphp/rector/2116)
<!-- Reviewable:end -->

TomasVotruba commented on README.md:
````diff
@@ -89,6 +89,10 @@ vendor/bin/rector process ../project --autoload-file ../project/vendor/autoload.
 
+```bash
+vendor/bin/rector process ../project --config ../project/rector.yaml
+```
````
The line above refers about the confit content, not `--config` option.

Better add use case bellow with example, when this option useful

voku commented:
@TomasVotruba you are right, so I added some more headlines and refactored many question in statement e.g. ```What can Rector do for You? -> Features``` || ```How to Apply Coding Standards? -> Rector !== Coding Standards```. I also added a simple link list on top and added the project logo. 

-> preview: https://github.com/voku/rector/tree/readme-update#rector

TomasVotruba commented:
I like current design, the menu is confusing even me.  
You can find tips for writing good readme: https://www.makeareadme.com/

Let's just add `--config` option.

voku commented:
Was only a suggestion. :-) Add the config example wherever you like. ;-)

## Configuration

#### Config Loading

The config file `rector.yaml` is passed to the rector executable with the `--config` option:

```bash
vendor/bin/rector process ../project --config ../project/rector.yaml
```

TomasVotruba commented:
That would be great
